pkg/controller/service/service_controller.go

```diff
@@ -115,8 +116,13 @@ func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Resul
 	// Check if this FlightRecorder already exists
 	found := &rhjmcv1alpha1.FlightRecorder{}
 	err = r.client.Get(ctx, types.NamespacedName{Name: jfr.Name, Namespace: jfr.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
+	if err != nil && kerrors.IsNotFound(err) {
 		reqLogger.Info("Creating a new FlightRecorder", "Namespace", jfr.Namespace, "Name", jfr.Name)
+		jmxPort, err := getServiceJMXPort(svc)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		jfr.Spec.Port = jmxPort
```

Comment:
We probably also need to handle the case where the port number is changed in the service after creation, and update the FlightRecorder accordingly.

Reply:
Good catch, I'll give that a shot too